registry/virtual.go

```diff
@@ -91,14 +508,18 @@ func OverwriteAppIcon(virtualSpaceName, appSlug, file string) error {
 		AppSlug:     appSlug,
 		ContentType: getMIMEType(file, []byte{}),
 	}
-	err = base.GlobalAssetStore.Add(a, icon, source)
-	if err != nil {
+	if err = base.GlobalAssetStore.Add(a, icon, source); err != nil {
 		return err
 	}
 	overwrite["icon"] = a.Shasum
 
 	id := getAppID(appSlug)
 	_, err = db.Put(context.Background(), id, overwrite)
```

Comment:
Shouldn't we check for an error before trying to regenerate overwritten tarballs?

Reply:
:up: